lang/fpc/Portfile

```diff
@@ -134,34 +133,50 @@ if {${subport} eq "${name}"} {
     }
 
     switch ${build_arch} {
-        "x86_64" {
-            set b "ppcx64"
+        x86_64 {
+            set bootstrapCompiler ppcx64
+            set cpuTarget x86_64
+            set compiler ppcx64
+        }
+        arm64 {
+            set bootstrapCompiler ppcx64
+            set cpuTarget aarch64
+            set compiler ppca64
         }
         default {
-            set b "unsupported"
+            set bootstrapCompiler unsupported
+            set cpuTarget unsupported
+            set compiler unsupported
         }
     }
 
+    # os.major = 20 is macOS = 11 (BigSur)
+    if {${os.major} >= 20} {
+        set linkerOptions -XR${configure.sdkroot}
+    } else {
```

Comment:
if you use ${configure.sdkroot} then every time there is a minor update to Xcode, this will break and need to be revbumped. See many similar tickets about many other ports that burn in the configure.sdkroot.

Reply:
if you change this linker flage to only apply on BigSur+, we will need to revbump them all again as that will change the way os.major < 20 links back to the way it was before, which is to use "/" preferentially I believe.

Reply:
I will take it back then.

Reply:
> I will take it back then.

Oh, it's a debate. I considered asking you to do it like this in the first place (then I would not have had to revbump all the older installs) but in the end,  the way you did it was simpler so I just left it and committed it.

Either way has merit -- on the one hand it is better to prefer fpc to test against and use the libraries in "/" if they exist, as that represents functions that the system actually supports. On the other, the newer SDK has more features and so will generally support builds longer -- although you run the risk of testing for libraries in the SDK, finding them, thinking they exist on the system, and then finding out they don't.

So long as you pick one or other, we'll be find. You're the maintainer of fpc now, so pick the one you feel is best, and just stick with it. Uncertainty and change are the main enemies here, not the choice you decide on.

Reply:
That gives me one more idea, at least as perspective. In case assembler and linker are available for i386 for all systems, that is all, what's needed for the i386 arch and with some extra effort even ppc. But let's get arm64 done first. Crosscompilers for windows are also an easy option as they use internal assembler and linker. Only Linux and related friends need external cross binutils.